### PR TITLE
update lesson, add source code files

### PR DESCRIPTION
**Pull request by linda-seiter**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -218,96 +370,136 @@ manually update all the previous data in the database.
...
+
+```java
+    @Transient
+    private String debugMessage;
+
+```
+
+## Primary Key Generation
+
+The `@GeneratedValue` annotation makes the database generate unique IDs
```
Comment (developerjaym) [inline README.md:385]: Line 393 in the code below has ```@Generated``` rather than ```@GeneratedValue```

File: README.md [hunk 2]
```diff
@@ -62,52 +77,53 @@ public class Student {
...
+because relying on the defaults may not provide the desired functionality.
 
 ### Dates
 
-We will remove the `age` property and add a `dob` (Date of Birth) property to
+Let's add a `dob` (Date of Birth) property to
 our `Student` class. The `@Temporal` annotation is used to define how the
 information will be stored in the database. We are using the `Date` class here
 to demonstrate the `@Temporal` annotation. For production apps, you would use
 the `LocalDate` class instead of the `Date` class.
```
Comment (developerjaym) [inline README.md]: If LocalDate is preferred for production apps, should we use that here instead, too?

Comment (kcleland0818) [reply README.md]: I agree with this statement.

**Review by kcleland0818 — APPROVED** [review]
Just a note to maybe include a reference to documentation about the `@GeneratedValue` annotation too if they are curious to learn more in the References section?

[https://docs.oracle.com/cd/E28389_01/apirefs.1111/e26376/javax/persistence/GeneratedValue.html](https://docs.oracle.com/cd/E28389_01/apirefs.1111/e26376/javax/persistence/GeneratedValue.html)

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md:385]: thanks, fixed.

File: README.md (hunk 2 above)
Comment (linda-seiter) [reply README.md]: updated to use LocalDate

**Review by kcleland0818 — APPROVED** [review]

**Review by developerjaym — APPROVED** [review]

**Review by developerjaym — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -178,7 +178,7 @@ public enum StudentGroup {
 
 The project structure should look like this:
 
-![project structure](https://curriculum-content.s3.amazonaws.com/6036/java-mod-5-jpa/studentgroup_enum.png)
+![project structure](https://curriculum-content.s3.amazonaws.com/6036/java-mod-5-jpa/studentgroup.png)
```
Comment (kcleland0818) [inline README.md]: Maybe crop the image a little more unless you want the file path to be kind of visible next to the `jpa-example` folder

File: README.md [hunk 4]
```diff
@@ -472,7 +472,7 @@ in increasing order:
 
 The project structure should look like this:
 
-![project structure](https://curriculum-content.s3.amazonaws.com/6036/java-mod-5-jpa/project_structure.png)
+![project structure](https://curriculum-content.s3.amazonaws.com/6036/java-mod-5-jpa/checkpoint_project.png)
```
Comment (kcleland0818) [inline README.md]: Same with this one?

**Review by kcleland0818 — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (linda-seiter) [reply README.md]: ok , updated.

File: README.md (hunk 3 above)
Comment (linda-seiter) [reply README.md]: ok, updated.

**Review by kcleland0818 — APPROVED** [review]

**Review by AlveeM — APPROVED** [review]

**Pull request merged**